apps/costcontrol/test/unit/formatting_test.js

```diff
@@ -200,26 +199,24 @@ suite('Formatting Test Suite >', function() {
       assert.equal(f.formatTimeSinceNow(now), 'just now');
       assert.equal(f.formatTimeSinceNow(aFewMinutesAgo), '25m ago');
       assert.equal(f.formatTimeSinceNow(aFewHoursAgo), '15h ago');
-      // As navigator.mozl10n.DateTimeFormat is a constructor, it is not
-      // possible to put an spy on its method localeFormat. For this reason,
-      // the spy must be placed into the Mock, instead of in the real method.
-      var dateSpy = this.sinon.spy(MockMozL10n.DateTimeFormat.prototype,
-                                   'localeFormat');
-      window.navigator.mozHour12 = true;
-      f.formatTimeSinceNow(oneTuesday);
-      assert.ok(dateSpy.calledTwice);
-      assert.ok(dateSpy.calledWith(oneTuesday, '%a'));
-      assert.ok(dateSpy.calledWith(oneTuesday,  'shortTimeFormat12'));
 
-      dateSpy.reset();
-      window.navigator.mozHour12 = false;
+      var calledTimes = 0;
+
+      var realDateTimeFormat = Intl.DateTimeFormat;
+
+      Intl.DateTimeFormat = function() {
```

Comment:
I would love to, but it doesn't work this day. `DateTimeFormat` doesn't work this way. `format` is not a method on a class, it's dynamically generated function and there's no way to spy on it :(
I tried with the new approach to set a spy on `Formatting.formats.shortTime.format` but that doesn't work either.
It seems that it's a getter, because `console.log(Formatting.formats.shortTime.format);` returns null, but executing this function works.

It's a more common problem and in many places in our code we have to deal with it.

Eventually, I realized that there's no need to test that. We are testing if Formatting methods return the right values, and it shouldn't be tested *how* they do this. The fact that Intl API is called twice inside it shouldn't matter if I'm correct as long as the returned value is correct.

So because with this change I am able to test the returned value against real data (instead of stubs of `mozL10n.DateTimeFormat`) I just removed that code.

Hope you'll like it:)